osu.Game/Rulesets/Scoring/HitWindows.cs

```diff
@@ -168,6 +182,29 @@ public double WindowFor(HitResult result)
         /// Defaults are provided but can be overridden to customise for a ruleset.
         /// </summary>
         protected virtual DifficultyRange[] GetRanges() => base_ranges;
+
+        public class EmptyHitWindows : HitWindows
+        {
+            private static readonly DifficultyRange[] ranges =
+            {
+                new DifficultyRange(HitResult.Perfect, 0, 0, 0),
+                new DifficultyRange(HitResult.Miss, 0, 0, 0),
+            };
+
+            public override bool IsHitResultAllowed(HitResult result)
+            {
+                switch (result)
+                {
+                    case HitResult.Great:
```

Comment:
Sorry, I meant this value should be `Perfect`
```suggestion
                    case HitResult.Perfect:
```